cohorts/utils.py

```diff
@@ -185,3 +185,10 @@ def get_logger(name, level=logging.INFO):
         logger.handlers = []
     logger.setLevel(level)
     return logger
+
+def set_attributes(obj, additional_data):
+    if additional_data is not None:
+        for key, value in additional_data.items():
+            if hasattr(obj, key):
+                raise ValueError("Key %s in additional_data already exists in this object" % key)
```

Comment:
Yeah, I think that's a great point and one I was/am a little worried about.

I wanted to error to make sure that we didn't accidentally override important variables, and it doesn't seem too hard for the user to remove or rename the input `DataFrame` to work around the error.

I don't feel strongly, but one option is to just put this in and see what errors we hit?